### PR TITLE
Fix #6375: Add miniphase for simple beta reductions

### LINKED ISSUE #6375: Double eta-expansion when ascribing contextual types
When we use a contextual type as a return type
```scala
class Test  {
   def f(): given Int => Boolean = true
}
```

it η-expands once ->

```scala
def f(): ImplicitFunction1[Int, Boolean] =  {
   def $anonfun(implicit evidence$1: Int): Boolean = true
   closure($anonfun)
}
```

However, if we provide an explicit ascription:

```scala
class Test  {
   def f(): given Int => Boolean = true : (given Int => Boolean)
}
```

it expands unnecessarily twice: 

```scala
def f(): ImplicitFunction1[Int, Boolean] =  {
   def $anonfun(implicit evidence$1: Int): Boolean =  (
      {
         def $anonfun(implicit evidence$2: Int): Boolean = true
         closure($anonfun)
      }:ImplicitFunction1[Int, Boolean]).apply(evidence$1)
   closure($anonfun)
}
```

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

**Comment by odersky** [comment]
test performance please

File: tests/pos/i6375.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/* In the following we should never have two nested closures after phase betaReduce
```
Comment (smarter) [inline tests/pos/i6375.scala]: We need a regression test for this in https://github.com/lampepfl/dotty/blob/master/compiler/test/dotty/tools/backend/jvm/InlineBytecodeTests.scala

Comment (odersky) [reply tests/pos/i6375.scala]: Yes, but I won't be the one adding it. Who can take care of this?

Comment (nicolasstucki) [reply tests/pos/i6375.scala]: Done.

@odersky could you double-check the bytecode to see it looks like what you expected.

**Pull request merged**